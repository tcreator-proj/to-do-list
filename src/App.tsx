import { Container, Row } from 'react-bootstrap';
import style from './app.module.css';
import ToDOWrapper from './components/ToDoWrapper';

function App() {
  return (
    <Container className={style.app}>
      <Row className={style['todo-container']}>
        <ToDOWrapper />
      </Row>
    </Container>
  );
}

export default App;
