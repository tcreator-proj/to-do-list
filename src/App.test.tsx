import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import userEvent from '@testing-library/user-event';

describe("App/ correct render", () => {
  it("find header text", async () => {
    render(<App />);
    expect(await screen.findByText(/to-do list/i)).toBeInTheDocument();
  })

  it("Find input field", async () => {
    render(<App />);
    expect(await screen.findByPlaceholderText(/What's need to be done?/i)).toBeInTheDocument()
  })

  it("Find mock block", () => {
    const {container} = render(<App />);
    expect(container.querySelector(".list-is-empty")).toBeInTheDocument()
  })
})

describe("App / Append task", () => {
  it("Create user task", async () => {
    render(<App />);

    userEvent.type(screen.getByRole('textbox'), "This is just task{enter}");
    expect(await screen.findByText(/This is just task/i)).toBeInTheDocument();
  })

  it("Create some user task and filter by completed", () => {
    render(<App />);

    userEvent.type(screen.getByRole('textbox'), "This is just task{enter}");
    userEvent.type(screen.getByRole('textbox'), "This is just task{enter}");
    userEvent.type(screen.getByRole('textbox'), "This is just task{enter}");

    userEvent.click(screen.getAllByRole('radio')[1]);

    expect(screen.queryByText(/This is just task/i)).not.toBeInTheDocument();
  })

  it("Create 1 user task, click it and filter by completed", () => {
    render(<App />);

    userEvent.type(screen.getByRole('textbox'), "This is just task{enter}");
    userEvent.click(screen.queryByText(/This is just task/i))

    userEvent.click(screen.getAllByRole('radio')[1]);

    expect(screen.queryByText(/This is just task/i)).toBeInTheDocument();
  })

  it("Create 3 user tasks, click one and clear by uncomplete", () => {
    const {container} = render(<App />);

    userEvent.type(screen.getByRole('textbox'), "First{enter}");
    userEvent.type(screen.getByRole('textbox'), "Second{enter}");
    userEvent.type(screen.getByRole('textbox'), "Third{enter}");
    userEvent.click(screen.queryByText(/First/i));

    userEvent.click(container.querySelector(".clear.col"));

    expect(container.querySelectorAll('.list-in-work').length).toEqual(2);
  })

  it("Create 3 user tasks, click some filters and check count on board", () => {
    const {container} = render(<App />);

    userEvent.type(screen.getByRole('textbox'), "First{enter}");
    userEvent.type(screen.getByRole('textbox'), "Second{enter}");
    userEvent.type(screen.getByRole('textbox'), "Third{enter}");
    userEvent.type(screen.getByRole('textbox'), "Fourth{enter}");
    userEvent.type(screen.getByRole('textbox'), "Fifth{enter}");
    userEvent.click(screen.queryByText(/First/i));
    userEvent.click(screen.queryByText(/Fourth/i));

    expect(container.querySelector(".task-count")?.textContent).toEqual('5')
    // userEvent.click(screen.queryByText(/Fourth/i));
    userEvent.click(screen.getAllByRole('radio')[1]);

    expect(container.querySelector(".task-count")?.textContent).toEqual('2');

    userEvent.click(screen.getAllByRole('radio')[2]);

    expect(container.querySelector(".task-count")?.textContent).toEqual('3');

    userEvent.click(screen.getAllByRole('radio')[0]);

    expect(container.querySelector(".task-count")?.textContent).toEqual('5');
  })
})
