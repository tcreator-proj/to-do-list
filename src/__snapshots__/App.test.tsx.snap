// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="app container"
  >
    <div
      class="todo-container row"
    >
      <div
        class="header row"
      >
        <h4
          class="head"
        >
          to-do list
        </h4>
      </div>
      <div
        class="row row"
      >
        <form
          class=""
        >
          <div>
            <input
              class="input form-control"
              id="task-input"
              max="100"
              min="1"
              name="inputValue"
              placeholder="What's need to be done?"
              type="text"
            />
          </div>
        </form>
      </div>
      <div
        class="panel row"
      />
      <div
        class="row"
      >
        <div
          class="list-is-empty"
        >
          ╮(￣_￣)╭
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
